spec/jobs/deposit_job_spec.rb

```diff
@@ -26,57 +42,133 @@
     allow(Honeybadger).to receive(:notify)
     # rubocop:disable RSpec/MessageChain
     allow(attached_file).to receive_message_chain(:file, :attachment, :blob).and_return(blob)
+    allow(attached_file2).to receive_message_chain(:file, :attachment, :blob).and_return(blob2)
     # rubocop:enable RSpec/MessageChain
-    allow(SdrClient::Deposit::UploadFiles).to receive(:upload)
-      .and_return([SdrClient::Deposit::Files::DirectUploadResponse.new(filename: 'sul.svg', signed_id: '9999999')])
   end
 
   after do
     blob.destroy
   end
 
-  context 'when the deposit request is successful' do
+  context 'when creating a new deposit is successful' do
     before do
       allow(SdrClient::Deposit::CreateResource).to receive(:run).and_return(1234)
+      allow(SdrClient::Deposit::UploadFiles).to receive(:upload)
+        .and_return([SdrClient::Deposit::Files::DirectUploadResponse.new(filename: 'sul.svg',
+                                                                         signed_id: '9999999')])
     end
 
-    it 'initiates a DepositStatusJob' do
-      described_class.perform_now(work_version)
+    it 'uploads files and calls CreateResource.run' do
+      described_class.perform_now(first_work_version)
+      expect(SdrClient::Deposit::CreateResource).to have_received(:run)
       expect(SdrClient::Deposit::UploadFiles).to have_received(:upload)
     end
 
     context 'when the deposit wants a doi' do
       let(:work) { build(:work, collection: collection, assign_doi: true) }
 
       it 'calls CreateResource.run with true for the assign_doi param' do
-        described_class.perform_now(work_version)
+        described_class.perform_now(first_work_version)
         expect(SdrClient::Deposit::CreateResource).to have_received(:run)
           .with(a_hash_including(accession: true, assign_doi: true))
       end
     end
   end
 
   context 'when updating the deposit' do
-    let(:work) { build(:work, collection: collection, assign_doi: false, druid: 'druid:bf024yb8975') }
+    let(:work) { build(:work, collection: collection, assign_doi: false, druid: druid) }
+    let(:druid) { 'druid:bf024yb8975' }
 
     before do
       allow(SdrClient::Deposit::UpdateResource).to receive(:run).and_return(1234)
     end
 
-    it 'calls UpdateResource.run' do
-      described_class.perform_now(work_version)
-      expect(SdrClient::Deposit::UpdateResource).to have_received(:run)
-        .with(a_hash_including(version_description: 'Updated metadata'))
+    context 'when files have not changed' do
+      # All that matters in this cocina is structural > contains.
```

Comment:
I don't understand this. Can you make it clearer?

Reply:
Additional comments added.